plutus-tx-plugin/src/PlutusTx/Compiler/Builtins.hs

```diff
@@ -346,6 +346,144 @@ defineBuiltinTerms = do
   defineBuiltinTerm annMayInline 'Builtins.bls12_381_G2_compressed_zero $
     PIR.mkConstant annMayInline BLS12_381.G2.compressed_zero
 
+  defineBuiltinTerm annMayInline 'Builtins.caseList' $
```

Comment:
We should be able to re-use what's defined in `PlutusCore.StdLib.Data.Data.matchData` and `PlutusCore.StdLib.Data.List.matchList`. Is there any reason we are duplicating it here?

Reply:
Besides, I think we need to remove `caseList` and `caseData` from `PlutusTx.Builtins.Internal` altogether since they are no longer "the special Haskell names that are used to map to builtin types or functions in Plutus Core." 

`caseList` and `caseData` should be moved to some other module in `PlutusTx` as a utility function imo